### PR TITLE
Add some permission utility methods.

### PR DESCRIPTION
**Pull request by tburry**
- Add PermissionModel->getPermissionsByRole().
- Add PermissionModel::addPermissions().
- Add Gdn_Session->addPermissions().

These methods are meant to aid plugin development that modifies the current user’s permissions in interesting ways.

### CODE REVIEW

File: applications/dashboard/models/class.permissionmodel.php [hunk 1]
```diff
@@ -70,6 +70,32 @@ public function addDefault($Type, $Permissions, $Junction = null, $JunctionId =
     }
 
     /**
+     * Add the permissions from one permission array to another.
+     *
+     * @param array $perms1 The permissions to be added to.
+     * @param array $perms2 The permissions to add.
+     * @return array Returns an array with all of the permissions in both permissions arrays.
+     */
+    public static function addPermissions($perms1, $perms2) {
```
Comment (linc) [inline applications/dashboard/models/class.permissionmodel.php]: Maybe `($currentPermissions, $newPermissions)` for the method signature? Numbered parameter names seems like bad precedent, no?

File: applications/dashboard/models/class.permissionmodel.php [hunk 2]
```diff
@@ -657,6 +683,40 @@ public function getPermissions($RoleID, $LimitToSuffix = '') {
...
+        $inc = Gdn::userModel()->getPermissionsIncrement();
+        $key = "perms:$inc:role:$roleID";
+
+        $permissions = Gdn::cache()->get($key);
+        if ($permissions === Gdn_Cache::CACHEOP_FAILURE) {
+            $sql = clone $this->SQL;
+            $sql->reset();
+
+            // Select all of the permission columns.
+            $PermissionColumns = $this->PermissionColumns();
```
Comment (linc) [inline applications/dashboard/models/class.permissionmodel.php]: Slipped back into StegoCase variables.

**Comment by nervoustwit** [comment]
I thought we were going to keep our comments in the imperative mood. Could you merge this?

File: applications/dashboard/models/class.permissionmodel.php (hunk 1 above)
Comment (tburry) [inline applications/dashboard/models/class.permissionmodel.php]: It's basically a merge and the 1, 2 etc is a precedent in the built in PHP functions.

**Pull request merged**